contracts/Compass.vy

```diff
@@ -21,6 +21,18 @@ interface ERC20:
     def transfer(_to: address, _value: uint256) -> bool: nonpayable
     def transferFrom(_from: address, _to: address, _value: uint256) -> bool: nonpayable
 
+interface FeeManager:
+    def deposit(depositor_paloma_address: bytes32): payable
+    def withdraw(receiver: address, amount:uint256, dex: address, payload: Bytes[1028], min_grain: uint256): nonpayable
+    def transfer_fees(fee_args: FeeArgs, relayer_fee: uint256, relayer: address): nonpayable
+    def security_fee_topup(): payable
+    def reserve_security_fee(sender: address, gas_fee: uint256): nonpayable
+    def bridge_community_fee_to_paloma(amount: uint256, dex: address, payload: Bytes[1028], min_grain: uint256) -> (address, uint256): nonpayable
```

Comment:
How static is the ABI for popular exchanges? Is there any way we can store it on compass and generate the needed bytecode for a swap on here?

I'm asking because otherwise we'll have to build a consensus layer on supported DEXes and their ABIs on Paloma in order to ensure this doesn't get called with the wrong parameters.

Reply:
Regarding this point, we should decide where to put liquidity for Grain-ETH. ABI is so different per dex. This pattern - dex & payload can work Uniswap v3, v2, or Curve.
So simple answer about static ABI for popular exchange is NO.
We can use different dex for exchange but we may use only one dex now until many grain holders put liquidity to different dex platforms. I will confirm this with @taariq 
And ABI is a kind of interface. In smart contract `interface` is used instead. Actually, a deployed smart contract on EVM has bytecode only and ABI makes it easier to run a tx / read data. Technically, we can interact with smart contracts without ABI.
However, in the pigeon module, you need to keep an ABI to generate a payload from the smart contract function name and parameters.